project_2/include/switch.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vector>
+#include <deque>
+#include "packet.hpp"
+#include "network_element.hpp"
+
+/**
+ * @brief Switch abstraction
+ * 
+ * @todo Think about situations when we dont know when packet will be sent next time
+ * 
+ */
+class NetworkSwitch: public RoutingNetworkElement {
+protected:
+    // TODO: add id (may be useful for more complicated topologies)
+    // Packet queue inside switch
+    // TODO: use dequeue instead of queue
+    std::deque<PacketHeader> buffer{};
+    std::uint64_t max_buffer_size_bit = 81920;
+    std::uint64_t current_buffer_size_bit = 0;
+    
+public:
+    /**
+     * @brief creates bottle neck in network and process packets
+     * 
+     * @param current_time_ns 
+     * @param packet 
+     * @param packets_wrapped 
```

Comment:
Nit: a common case when Doxygen doc string deviates from the actual implementation. Therefore, let us not use Doxygen at all in the `main` branch code to simplify the code maintenance.